commands/proposal_submission.go

```diff
@@ -375,35 +379,54 @@ func checkOracleSpec(spec *oraclespb.OracleSpecConfiguration, name string) Error
 				errs.AddForProperty(fmt.Sprintf("proposal_submission.terms.change.new_market.changes.instrument.product.future."+name+".pub_keys.%d", i), ErrIsNotValid)
 			}
 		}
-		if len(spec.Filters) == 0 {
-			errs.AddForProperty("proposal_submission.terms.change.new_market.changes.instrument.product.future."+name+".filters", ErrIsRequired)
-		} else {
-			for i, filter := range spec.Filters {
-				if filter.Key == nil {
-					errs.AddForProperty(fmt.Sprintf("proposal_submission.terms.change.new_market.changes.instrument.product.future."+name+".filters.%d.key", i), ErrIsNotValid)
-				} else {
-					if len(filter.Key.Name) == 0 {
-						errs.AddForProperty(fmt.Sprintf("proposal_submission.terms.change.new_market.changes.instrument.product.future."+name+".filters.%d.key.name", i), ErrIsRequired)
-					}
-					if filter.Key.Type == oraclespb.PropertyKey_TYPE_UNSPECIFIED {
-						errs.AddForProperty(fmt.Sprintf("proposal_submission.terms.change.new_market.changes.instrument.product.future."+name+".filters.%d.key.type", i), ErrIsRequired)
-					}
+
+		return checkOracleSpecFilters(spec, name, errs)
+	} else {
+		errs.AddForProperty("proposal_submission.terms.change.new_market.changes.instrument.product.future."+name, ErrIsRequired)
+	}
+
+	return errs
+}
+
+func isBuiltInSpec(filters []*oraclespb.Filter) bool {
+	if len(filters) == 0 || len(filters) > 1 {
+		return false
+	}
+
+	if strings.HasPrefix(filters[0].Key.String(), "vegaprotocol.builtin") {
+		return true
+	}
+
+	return false
+}
+
+func checkOracleSpecFilters(spec *oraclespb.OracleSpecConfiguration, name string, errs Errors) Errors {
+	if len(spec.Filters) == 0 {
+		errs.AddForProperty("proposal_submission.terms.change.new_market.changes.instrument.product.future."+name+".filters", ErrIsRequired)
```

Comment:
I think, you can return directly after this check, and thus remove the `else` branch so we can minimise indentation :)

Reply:
Yes, missed that. Thank you.

Reply:
Resolved https://github.com/vegaprotocol/protos/pull/253/commits/4fc21d339ad4e84a4aaf4df70a1ff481392db2ee